stripe-ui-core/src/main/java/com/stripe/android/uicore/elements/TextFieldController.kt

```diff
@@ -175,4 +177,12 @@ class SimpleTextFieldController constructor(
     override fun onFocusChange(newHasFocus: Boolean) {
         _hasFocus.value = newHasFocus
     }
+
+    private fun TextFieldState.isComplete() =
+        isValid() || (!isValid() && showOptionalLabel && isBlank())
+
+    private data class TextField(
+        val value: String,
+        val state: TextFieldState
+    )
```

Comment:
The class names get a little messy here. Would it makes sense to rename them?

- `TextField` could become `TextFieldState`
- `TextFieldState` could become `TextFieldValidity` (or any other suitable term)